pkg/route/route.go

```diff
@@ -169,10 +169,11 @@ func forgeNetlinkRoute(route *networkingv1alpha1.Route, tableID uint32) (*netlin
 
 	if route.Dev != nil {
 		link, err := netlink.LinkByName(*route.Dev)
-		if err != nil {
-			return nil, err
+		if err == nil {
+			linkIndex = link.Attrs().Index
+			// We don't return an error if the device is not found.
+			// This is because the device could be absent when the route is being deleted.
```

Comment:
Not necessary, because in the deletion of a route we only use the destination CIDR/IP to select the route

Reply:
I decided to maintain the error return in case the dev does not exist. I removed the forgeNetlinkRule function from EnsureRouteAbscence and I swaped it with this https://github.com/cheina97/liqo/blob/04401f40cbf06618570c123d921dfe15f2130769/pkg/route/route.go#L55-L64